plugins/module_utils/elbv2.py

```diff
@@ -1217,6 +1224,9 @@ def create(self):
         try:
             self.rule["ListenerArn"] = self.listener_arn
             self.rule["Priority"] = int(self.rule["Priority"])
+            for action in self.rule.get("Actions", []):
+                if action.get("AuthenticateOidcConfig", {}).get("UseExistingClientSecret", False):
```

Comment:
I wonder if we could move this (and the similar addition in the `modify` method below) to happen in the `compare_rules` method instead so that the returned `rules_to_add` and `rules_to_modify` contain the final expected data to send? That way the new unit tests could verify this behavior, which I don't believe is currently being tested.